### PR TITLE
Report host+port for discovered casts

### PR DESCRIPTION
**Pull request by emontnemery**
Report host+port for discovered casts
Add new discovery helper `discover_listed_chromecasts`
Add additional discovery examples

Fix default value of zconf in Chromecast constructor

### CODE REVIEW

File: examples/discovery_example2.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+parser = argparse.ArgumentParser(description="Example on how to receive updates on discovered chromecasts.")
+parser.add_argument("--show-debug", help="Enable debug log", action="store_true")
+args = parser.parse_args()
+
+if args.show_debug:
+    logging.basicConfig(level=logging.DEBUG)
+
+devices, browser = pychromecast.discovery.discover_chromecasts()
+# Shut down discovery
+pychromecast.stop_discovery(browser)
```
Comment (tareco) [inline examples/discovery_example2.py:20]: I am getting this:
AttributeError: module 'pychromecast' has no attribute 'stop_discovery'

Comment (emontnemery) [reply examples/discovery_example2.py:20]: Did you include all the changes from the PR?
`stop_discovery` is readded to pychromecast here: https://github.com/home-assistant-libs/pychromecast/blob/b43127e0e6d884356ef3622d72a468e2eb544346/pychromecast/__init__.py#L21

File: pychromecast/discovery.py [hunk 2]
```diff
@@ -166,6 +171,47 @@ def callback(uuid, name):
     return (listener.devices, browser)
 
 
+def discover_listed_chromecasts(
```
Comment (balloob) [inline pychromecast/discovery.py:174]: One more discovery method 😆 Should we remove some other ones?

File: examples/discovery_example2.py (hunk 1 above)
Comment (tareco) [reply examples/discovery_example2.py:20]: oops.. you're right :P

File: pychromecast/discovery.py (hunk 2 above)
Comment (emontnemery) [reply pychromecast/discovery.py:174]: I think it's fairly OK:
 - `discover_chromecasts` and `discover_listed_chromecasts` are helpers for discovery only.
 - `get_chromecasts` and `get_listed_chromecasts` are discovers and creates Chromecast objects.

**Pull request merged**